### PR TITLE
Delete Hardhat

### PR DESCRIPTION
**Pull request by ewilz**
Currently accommodating for both hardhat and foundry can be a lot of tech debt. removing hardhat for now.

Final todos:
- [ ]  migrate PoolManager.gas.spec.ts

### CODE REVIEW

**Review by zhongeric — DISMISSED** [review]

File: test/foundry-tests/PoolManager.t.sol [hunk 1]
```diff
@@ -1023,6 +1039,213 @@ contract PoolManagerTest is Test, Deployers, TokenFixture, GasSnapshot, IERC1155
...
+        uint256 expectedFees = 7;
+
+        PoolKey memory key =
+            PoolKey({currency0: currency0, currency1: currency1, fee: 3000, hooks: IHooks(address(0)), tickSpacing: 10});
+        manager.setProtocolFeeController(IProtocolFeeController(address(feeController)));
+        feeController.setSwapFeeForPool(key.toId(), uint16(protocolFee));
+
+        manager.initialize(key, SQRT_RATIO_1_1, ZERO_BYTES);
+        (Pool.Slot0 memory slot0,,,) = manager.pools(key.toId());
+        assertEq(slot0.protocolFees, protocolFee << 12);
```
Comment (snreynolds) [inline test/foundry-tests/PoolManager.t.sol:1140]: would add the assertion that the withdraw protocol fee is 0

**Review by snreynolds — CHANGES_REQUESTED** [review]

File: test/PoolManager.gas.spec.ts [hunk 2]
```diff
@@ -1,566 +1,566 @@
...
+//
+//       it('second swap in block, large swap crossing several initialized ticks', async () => {
+//         await modifyPosition(startingTick - 3 * tickSpacing, startingTick - tickSpacing, expandTo18Decimals(1))
+//         await modifyPosition(startingTick - 4 * tickSpacing, startingTick - 2 * tickSpacing, expandTo18Decimals(1))
+//         await swapExact0For1(expandTo18Decimals(1).div(10000), wallet.address)
+//         await snapshotGasCost(swapExact0For1(expandTo18Decimals(1), wallet.address))
+//         expect((await getSlot0()).tick).to.be.lt(startingTick - 4 * tickSpacing)
+//       })
+//
+//       it('second swap in block, large swap crossing a single initialized tick', async () => {
```
Comment (hensha256) [inline test/PoolManager.gas.spec.ts:523]: for what reason is this file being kept?

**Review by hensha256 — APPROVED** [review]
lgtmmm 🚀

File: test/foundry-tests/PoolManager.t.sol [hunk 3]
```diff
@@ -1036,6 +1052,213 @@ contract PoolManagerTest is Test, Deployers, TokenFixture, GasSnapshot, IERC1155
...
+
+        assertEq(manager.protocolFeesAccrued(currency0), expectedFees);
+        assertEq(manager.protocolFeesAccrued(currency1), 0);
+        assertEq(currency0.balanceOf(address(1)), 0);
+        manager.collectProtocolFees(address(1), currency0, 0);
+        assertEq(currency0.balanceOf(address(1)), expectedFees);
+        assertEq(manager.protocolFeesAccrued(currency0), 0);
+    }
+
+    function test_collectProtocolfees_nativeToken_allowsOwnerToAccumulateFees() public {
```
Comment (hensha256) [inline test/foundry-tests/PoolManager.t.sol:1196]:
```suggestion
    function test_collectProtocolFees_nativeToken_allowsOwnerToAccumulateFees() public {
```

Comment (hensha256) [reply test/foundry-tests/PoolManager.t.sol:1196]: on other tests too

File: test/PoolManager.gas.spec.ts (hunk 2 above)
Comment (snreynolds) [reply test/PoolManager.gas.spec.ts:523]: We havent moved over these tests so I think she wanted to keep them but doesnt want to run them :)

Comment (zhongeric) [reply test/PoolManager.gas.spec.ts:523]: ya lets keep for reference

File: test/foundry-tests/PoolManager.t.sol (hunk 1 above)
Comment (zhongeric) [reply test/foundry-tests/PoolManager.t.sol:1153]: todo?

**Review by zhongeric — DISMISSED** [review]

**Comment by tom2o17** [comment]
Booo Hardat + Foundry >>>>

**Review by zhongeric — APPROVED** [review]

**Review by snreynolds — APPROVED** [review]

**Pull request merged**